tilelang/autodd.py

```
@@ -0,0 +1,1156 @@
+
+    def stop_proc(self):
+        pass
+
+    async def run(self, code: str, timeout: float = 5.0):
+        with tempfile.NamedTemporaryFile("w", suffix=".py", delete=True) as f:
+            f.writelines(code)
+            f.flush()
+
+            def run_subprocess(args):
+                try:
+                    proc = subprocess.run(
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)